src/t_string.c

```diff
@@ -571,8 +571,8 @@ void msetnxCommand(client *c) {
 void incrDecrCommand(client *c, long long incr) {
     long long value, oldvalue;
     robj *o, *new;
-
-    o = lookupKeyWrite(c->db,c->argv[1]);
+    dictEntry *de = dbFind(c->db, c->argv[1]->ptr);
+    o = lookupKeyWriteWithDictEntry(c->db,c->argv[1],de);
```

Comment:
make lookupKeyWriteWithDictEntry() returns the current de, rather than the caller incoming?

Reply:
@sundb we need the replied `robj* o`.